lib/game_client/ogs/ogs_game.dart

```diff
@@ -433,18 +453,41 @@ class OGSGame extends Game {
   void _handleClock(Map<String, dynamic> data) {
     _logger.fine('Received clock update for game $id');
 
+    // If the game has already ended, stop both timers and ignore clock updates
+    if (_resultCompleter.isCompleted) {
+      _blackTimer.stop();
+      _whiteTimer.stop();
+      _logger.fine('Ignoring clock update for completed game $id');
+      return;
+    }
+
     try {
       final blackTimeData = data['black_time'] as Map<String, dynamic>?;
       final whiteTimeData = data['white_time'] as Map<String, dynamic>?;
+      final currentPlayer = data['current_player'] as int?;
 
       if (blackTimeData != null) {
-        blackTime.value =
-            (blackTime.value.$1 + 1, _parseOGSTimeData(blackTimeData));
+        final newState = _parseOGSTimeData(blackTimeData);
+        final isBlackTurn = currentPlayer == int.tryParse(black.value.userId);
+
+        if (isBlackTurn) {
+          _blackTimer.start(newState);
+        } else {
+          _blackTimer.stop();
+          blackTime.value = (blackTime.value.$1 + 1, newState);
+        }
       }
 
       if (whiteTimeData != null) {
-        whiteTime.value =
-            (whiteTime.value.$1 + 1, _parseOGSTimeData(whiteTimeData));
+        final newState = _parseOGSTimeData(whiteTimeData);
```

Comment:
a bit of duping here, maybe we should refactor it into its own function? (it can take the timer and the listener as arguments)

Reply:
SGTM.  The code isn't much shorter, but I agree the de-duped logic is better for maintenance.